api/src/opentrons/protocol_engine/state/commands.py

```diff
@@ -181,7 +181,9 @@ def handle_action(self, action: Action) -> None:  # noqa: C901
             # For now, unit tests cover mapping every request type
             queued_command = action.request._CommandCls.construct(
                 id=action.command_id,
-                key=action.command_key,
+                key=action.request.key
+                if action.request.key is not None
+                else action.command_id,
```

Comment:
Can we use parentheses for readability and preserve the TODO from `protocol_engine.py`?

```suggestion
                key=(
                    action.request.key
                    if action.request.key is not None
                    # TODO(mc, 2021-12-13): generate a command key from params and state
                    # https://github.com/Opentrons/opentrons/issues/8986
                    else action.command_id
                ),
```